logstash-core/src/main/java/org/logstash/config/ir/graph/Graph.java

```diff
@@ -269,7 +269,7 @@ public Integer rank(Vertex vertex) {
     }
 
     public Map<String, List<Vertex>> verticesByHash() {
-        return this.vertices().collect(Collectors.groupingBy(Vertex::uniqueHash));
+        return this.vertices().parallel().collect(Collectors.groupingBy(Vertex::uniqueHash));
```

Comment:
@andrewvc why parallel? The stream you get back won't be `CONCURRENT`will it?
<img width="554" alt="screen shot 2017-06-10 at 21 24 32" src="https://user-images.githubusercontent.com/6490959/27005682-92921692-4e23-11e7-94cf-9978dd6d9f9f.png">